### PR TITLE
feat: add support for copying a referrer link

### PR DESCRIPTION
**Pull request by ssbarbee**
* consume heroku database to copy the referrer link
* testing url: https://feat-consume-database-to-generate-invite-link.d2f2fhja7tsifl.amplifyapp.com/#/trader-pools

### CODE REVIEW

File: src/routes/Profile/get-referrer-link.ts
Comment (CyclopsRex) [inline src/routes/Profile/get-referrer-link.ts]: delete?

File: src/routes/Profile/get-referrer-link.ts
Comment (CyclopsRex) [inline src/routes/Profile/get-referrer-link.ts:4]: As discussed I am defaulting this in https://github.com/Voltz-Protocol/voltz-ui/pull/375 if not set, cos that seemed better than errors or weird behaviour that we might not catch in testing. Totally open to not defaulting but we should probably be consistent.

File: src/routes/Profile/get-referrer-link.ts
Comment (CyclopsRex) [inline src/routes/Profile/get-referrer-link.ts:22]: I fixed the API to return 404 properly if not found so if you wrote/tested this code before that then the error handling here may need to change.

File: src/routes/Profile/Profile.tsx
Comment (CyclopsRex) [inline src/routes/Profile/Profile.tsx:215]: Do I understand correctly that this only gets called when they click the link? Could take a few hundred ms to get code from API in the worst case, so is there any value in pre-fetching the code when we load the profile page?

File: src/routes/Profile/get-referrer-link.ts
Comment (CyclopsRex) [inline src/routes/Profile/get-referrer-link.ts]: A quick test suggests that this doesn't work with https://github.com/Voltz-Protocol/voltz-ui/pull/375, which was tested with URIs of the form http://localhost:3000/#/trader-pools?invitedBy=SWAEUlit cos that's what we used in the hackathon. If you think this URL without the `/#/trader-pools` part is better, then please can you suggest a new home for the localStorage-writing code in https://github.com/Voltz-Protocol/voltz-ui/pull/375 ?

**Review by CyclopsRex — COMMENTED** [review]
Some comments below. Wonder if we should merge this with https://github.com/Voltz-Protocol/voltz-ui/pull/375 and test them together before merging into develop. But I'm out tomorrow so I'll leave this up to you.

File: src/routes/Profile/get-referrer-link.ts
Comment (ssbarbee) [reply src/routes/Profile/get-referrer-link.ts]: fixed in 5e7aa349877b2a2309a6a6e6afbe1474d8740b48

File: src/routes/Profile/get-referrer-link.ts
Comment (ssbarbee) [reply src/routes/Profile/get-referrer-link.ts:4]: Me and @0xZenus created ticket on this matter. We will have a look into this in separate ticket addressing most of ENVs

File: src/routes/Profile/Profile.tsx
Comment (ssbarbee) [reply src/routes/Profile/Profile.tsx:215]: I don't want to "create" network calls if not needed. Create them per request of user. Potentially I can add caching if the same "code" is returned always

File: src/routes/Profile/get-referrer-link.ts
Comment (ssbarbee) [reply src/routes/Profile/get-referrer-link.ts]: yes I will fix this in #375 and test it accordingly. Thanks for pointing it out.

Comment (CyclopsRex) [reply src/routes/Profile/get-referrer-link.ts]: Great, thanks! Please resolve this conversation once fixed in #375

File: src/routes/Profile/Profile.tsx
Comment (CyclopsRex) [reply src/routes/Profile/Profile.tsx:215]: Same code is always returned for a given user.

**Review by 0xZenus — APPROVED** [review]

**Pull request merged**
